### ISSUE TITLE
TemplateClient.Validate in armresource pkg is not functional on error case

### ISSUE THREAD

**Issue by haitch** [Azure/azure-sdk-for-go#21357]
### Bug Report
- import path of package in question, ```"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/resources/armresources```
- SDK version: ```all``` (I believe)
- output of `go version` 1.20 (doesn't matter)

- What happened?
given following code

- What did you expect or want to happen?
- How can we reproduce it?
- Anything we should know about your environment.

**Comment by jhendrixMSFT** [comment]
OpenAPI for this operation https://github.com/Azure/azure-rest-api-specs/blob/main/specification/resources/resource-manager/Microsoft.Resources/stable/2021-04-01/resources.json#L1331

**Comment by haitch** [comment]
so these 2 condition should be combined together, we need to kick out one of them, I vote to kick out ```"x-ms-long-running-operation": true```, at least for this APIVersion.

**Comment by JeffreyRichter** [comment]
Sidenote for @jhendrixMSFT : I see runtime.PollUntilDoneOptions is useful to customers and in the runtime package - we should consider moving it.

I found the REST API docs (certainly created from the swagger) for this here: https://learn.microsoft.com/en-us/rest/api/resources/deployments/validate-at-subscription-scope#deploymentproperties. The docs are so bad, I'd be surprised if anyone could use this correctly.

If this is supposed to be an LRO, then there are no docs describing how to poll for the status. For example, no mention of a response header with a URL to use for polling.
The docs indicating that 400 returns a DeploymentValidateResult seems anti-HTTP and bad design. 400 means that the operation did not work at all.

We'd have to get @rkmanda and the service team involved to at least improve the docs and probably the swagger so it reflects how the operation actually works today. Then maybe, we fix how the operation works to make it follow HTTP (and Azure) guidelines so it works as people might expect it to.

**Comment by haitch** [comment]
> The docs indicating that 400 returns a DeploymentValidateResult seems anti-HTTP and bad design. 400 means that the operation did not work at all.
exactly, this should be a 200 response, with some response payload, that describe validation result.

**Comment by sushmak295** [comment]
any update on this is highly appreciated

**Comment by sushmak295** [comment]
any update on this is highly appreciated

**Comment by olegch** [comment]
This affects us as well.
It seems that all `BeginValidate*` methods have the same issue, at least we are facing it with `BeginValidate`
